Add M524 to abort SD printing

TheSFReader opened the pull request:
This is a port to 2.0.x of  PR #11386 by @zachwelch. 

"It adds the M524 command as "SD abort job", performing the same logical checks that enable the Stop SD Job option in the LCD menu."

TheSFReader committed:
Add M524 to abort SD printing

Co-Authored-By: Zach Welch <zachwelch@users.noreply.github.com>

thinkyhead commented:
Have you added this G-code to the RepRap wiki and to MarlinDocumentation yet?

thinkyhead commented:
Note that the commit Summary message should be quite short. If extended description is needed, please use the Description region.

thinkyhead committed:
Move abort_sd_printing to cardreader class

(Otherwise the `extern` declaration should be in `Marlin.h`, not `cardreader.h`)

Co-Authored-By: Zach Welch <zachwelch@users.noreply.github.com>
Co-Authored-By: TheSFReader <thesfreader@gmail.com>

TheSFReader commented:
Nope, not yet added, will try and do it now. sorry about the commit message

thinkyhead commented:
No worries. I've just added the codes to the wiki and MarlinDocumentation, so we're all set!

TheSFReader commented:
Thanks ! Just started cloning the MarlinDocumentation ... Will use next time ;)
